### PR TITLE
fix(datetime): add keyboard year navigation

### LINKED ISSUE #18122: feat: Use arrows to change day/month/year HH:SS in datetime
# Feature Request

**Ionic version:**

[x] **4.3.0**

**Describe the Feature Request**

Currently `ion-datetime` control is accessible through keyboard in non optimal way. To change day/month user should use `TAB` key to go to target value and then press `space` key to select it.

The issue with this is that if user needs to change just value in 2nd column (s)he needs to `TAB` skip all the values in 1st one (e.g. 31 days for example). This is very daunting experience. It's also does not help that while tab-jumping though values the current column is not scrolled so user can't see currently focused item when it's out of visible window.

**Describe Preferred Solution**

The better solution would be to use left-right keys to change current columns and up-down keys to navigate values. The space still can be used to select focused value to be current.

The tab key should be used/reserved to navigation between controls. So when in datetime control and `tab` pressed the datetime should close and next control should get focus.

**Describe Alternatives**

Also it would be super cool to support entering date with number keys.

**Related Code**

http://plnkr.co/edit/li6pVumtGoVGrhjTbAqg?p=preview

**Additional Context**

### LINKED ISSUE #21553: bug: Ion-datetime component is not accessible for the user using keyboard
# Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.x**

**Current behavior:**
The screen reads the ion-datetime component as a button. It’s not reading the associate aria-label or the <ion-label> for it. Also, the date selector is not accessible using only keyboard. It does not receive the focus by default.

Issue is reproducible from the sample implementation also
https://ionicframework.com/docs/demos/api/datetime/index.html?ionic:mode=ios

Refer to item 4 in the document shared in Slack for further details.
https://ionicteam.slack.com/archives/C7CGXAVTL/p1592319616186200

### PR DESCRIPTION
**Pull request by willmartindev**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: 
- resolves #21553 
- resolves #18122

## What is the new behavior?

- Users can now navigate years in `ion-datetime` by pressing (`Shift` + `PageUp`) or (`Shift` + `PageDown`).
- Added accessibility section to `ion-datetime` docs.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Keyboard behavior based on https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/datepicker-dialog.html

### CODE REVIEW

File: core/src/components/datetime/utils/manipulation.ts [hunk 1]
```diff
@@ -256,6 +256,30 @@ export const getNextMonth = (refParts: DatetimeParts) => {
...
+ */
+export const getPreviousYear = (refParts: DatetimeParts) => {
+  return changeYear(refParts, -1);
+}
+
+/**
+ * Given DatetimeParts, generate the next year.
+ */
+export const getNextYear = (refParts: DatetimeParts) => {
+  return changeYear(refParts, +1);
```
Comment (liamdebeasi) [inline core/src/components/datetime/utils/manipulation.ts:280]:
```suggestion
  return changeYear(refParts, 1);
```

Minor style thing, but the positive here is implied

**Review by liamdebeasi — CHANGES_REQUESTED** [review]
Other than a small style comment this looks good to go

File: core/src/components/datetime/utils/manipulation.ts (hunk 1 above)
Comment (willmartindev) [reply core/src/components/datetime/utils/manipulation.ts:280]: thanks!

**Review by liamdebeasi — APPROVED** [review]
Nice job!

**Pull request merged**